saltapi/service/institution_service.py

```diff
@@ -10,3 +11,10 @@ def __init__(self, repository: InstitutionRepository):
     def get_institutions(self) -> List[Dict[str, Any]]:
         institutions = self.repository.get_institutions()
         return institutions
+
+    def create(self, new_institution_details: NewInstitutionDetails) -> None:
+        self.repository.create(new_institution_details)
+
+    def get_institution_by_name(self, institution_name: str) -> Institution:
```

Comment:
See the comment on uniqueness in the repository file; it's rather the combination of names and department that should be unique.